rskj-core/src/test/java/co/rsk/core/bc/TransactionPoolImplTest.java

```diff
@@ -512,8 +514,10 @@ public void addTwiceAndGetQueuedTransaction() {
         transactionPool.addTransaction(tx);
 
         TransactionPoolAddResult result = transactionPool.addTransaction(tx);
-        Assert.assertFalse(result.transactionWasAdded());
-        result.ifTransactionWasNotAdded(msg -> Assert.assertEquals("queued transaction with same hash already exists", msg));
+        Assert.assertFalse(result.transactionsWereAdded());
+        if(!result.transactionsWereAdded()) {
```

Comment:
You should remove this if, it is confusing, you already asserted it was false the line above

Reply:
yeah it sounds weird but in this way i'm asserting error message

Reply:
removed unnecesary if